x-pack/plugins/actions/server/builtin_action_types/connectors/schema.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const MappingActionType = schema.oneOf([
+  schema.literal('nothing'),
+  schema.literal('overwrite'),
+  schema.literal('append'),
+]);
+
+export const MapRecordSchema = schema.object({
+  source: schema.string(),
+  target: schema.string(),
+  actionType: MappingActionType,
+});
+
+export const CaseConfigurationSchema = schema.object({
+  mapping: schema.arrayOf(MapRecordSchema),
+});
+
+export const ConnectorPublicConfiguration = {
+  apiUrl: schema.string(),
+  casesConfiguration: CaseConfigurationSchema,
```

Comment:
nit: why `casesConfiguration` and not `caseConfiguration`

Reply:
As far as I remember I choose this name because the configuration concerns cases in general and not a particular case. If you think is better to change it please let me know.